File: Javascript-location/domain/Reservation.js
class Reservation {
  constructor(item, startDate, durationDays) {
    if (durationDays < 1) {
      throw new Error("Durée minimum : 1 jour.");
    }

    if (!item.isAvailable(startDate, durationDays)) {
      throw new Error("L'objet est déjà réservé à cette période.");
    }

    this.item = item;
    this.startDate = startDate;
    this.durationDays = durationDays;

    item.addReservation(startDate, durationDays);
  }
}

module.exports = Reservation;

File: python-location/domain/reservation.py
class Reservation:
    def __init__(self, item, start_date, end_date):
        if (end_date - start_date).days < 1:
            raise ValueError("Durée minimum : 1 jour.")

        if not item.is_available(start_date, end_date):
            raise ValueError("L'objet est déjà réservé à cette période.")

        self.item = item
        self.start_date = start_date
        self.end_date = end_date

        item.add_reservation(start_date, end_date)